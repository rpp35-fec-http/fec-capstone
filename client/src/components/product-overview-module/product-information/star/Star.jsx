import * as React from 'react';
import './Star.scss';

const SvgComponent = (props) => (
  <svg
    viewBox={`0 0 ${props.rating} 50`}
    xmlns='http://www.w3.org/2000/svg'
    id='star-container'
    xmlSpace='preserve'
    style={{
      fillRule: 'evenodd',
      clipRule: 'evenodd',
      strokeLinejoin: 'round',
      strokeMiterlimit: 2,
    }}
    {...props}
  >
    <path
      d='M0 37.069h102L19.491 96.991 51.041.009l31.55 96.982L0 37.069Z'
      style={{
        fill: '#ff0',
        fillRule: 'nonzero',
      }}
      transform='matrix(.36274 0 0 .36274 7 6.907)'
    />
  </svg>
);

export default SvgComponent;
