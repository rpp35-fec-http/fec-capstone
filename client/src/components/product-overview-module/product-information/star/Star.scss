@use '../../../App.scss' as *;

* {
  margin: 0;
  padding: 0;
}

#star-container {
  // height: 50px;
  transform: scale(1);
  // @include set-size(60px, 60px);
  animation: star 1s ease;
  transform-origin: center;
  transform-box: fill-box;
}

@keyframes star {
  from {
    transform: rotateZ(0deg);
  }
  to {
    transform: rotateZ(360deg);
  }
}
