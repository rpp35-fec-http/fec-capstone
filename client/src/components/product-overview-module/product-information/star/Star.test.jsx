import renderer from 'react-test-renderer';
import Star from './Star.jsx';

it('renders correctly', () => {
   const tree = renderer.create(<Star />).toJSON();
   expect(tree).toMatchSnapshot();
});
