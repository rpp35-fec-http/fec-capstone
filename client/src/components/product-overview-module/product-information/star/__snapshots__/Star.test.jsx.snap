// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<svg
  height="100%"
  id="star"
  style={
    Object {
      "clipRule": "evenodd",
      "fillRule": "evenodd",
      "strokeLinecap": "round",
      "strokeLinejoin": "round",
      "strokeMiterlimit": 1.5,
    }
  }
  viewBox="0 0 800 600"
  width="100%"
  xmlSpace="preserve"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <g
    transform="matrix(1.40897,0,0,1.40897,-67.8922,-236.585)"
  >
    <path
      d="M333.5,174L384.352,330.508L548.914,330.508L415.781,427.235L466.633,583.742L333.5,487.015L200.367,583.742L251.219,427.235L118.086,330.508L282.648,330.508L333.5,174Z"
      style={
        Object {
          "fill": "none",
          "stroke": "black",
          "strokeWidth": "5.68px",
        }
      }
    />
  </g>
</svg>
`;
